src/libraries/System.Net.Http/tests/FunctionalTests/MetricsTest.cs

```diff
@@ -617,24 +617,30 @@ public async Task RequestDuration_EnrichmentHandler_ContentLengthError_Recorded(
         [Fact]
         public async Task FailedRequests_ConnectionClosedWhileReceivingHeaders_Recorded()
         {
-            TimeSpan timeout = TimeSpan.FromSeconds(30);
             await LoopbackServerFactory.CreateClientAndServerAsync(async uri =>
             {
                 using HttpMessageInvoker client = CreateHttpMessageInvoker();
                 using InstrumentRecorder<long> recorder = SetupInstrumentRecorder<long>(InstrumentNames.FailedRequests);
                 using HttpRequestMessage request = new(HttpMethod.Get, uri) { Version = UseVersion };
 
-                await Assert.ThrowsAsync<HttpRequestException>(async () =>
+                Exception ex = await Assert.ThrowsAnyAsync<Exception>(async () =>
                 {
-                    using HttpResponseMessage response = await SendAsync(client, request);
-                }).WaitAsync(timeout);
+                    // Getting a cancellation is also good if we are unable to detect the peer shutdown.
+                    using CancellationTokenSource cts = new CancellationTokenSource(1000);
```

Comment:
preferably, we should not use magic numbers. Can we find some known passing/failing constant? .

Reply:
We only have `TestHelper.PassingTestTimeoutMilliseconds = 60sec`. This is too much, if we don't receive an RST from the server, I want it to fire relatively quickly while still making sure `MetricsHandler` is reached. Changed it to 10sec. I don't think we have constants with similar semantics.